test/internal/cmd/io.go

```diff
@@ -38,12 +41,45 @@ func (b *BufferedIO) Output() (_ string, err error) {
 func (b *BufferedIO) TestOutput(tb testing.TB, out string, err error) {
 	tb.Helper()
 
-	outGive, errGive := b.Output()
-	if !errors.Is(errGive, err) {
-		tb.Fatalf("got stderr: %v; wanted: %v", errGive, err)
+	outGot, errGot := b.Output()
+	if !errors.Is(errGot, err) {
```

Comment:
If `errGot` is say `errors.New("file not found")`  will it match with `os.ErrNotExist`?

Reply:
nope, `errors.Is` checks that it matches the original error object, not the message itself

Reply:
Isn't that a problem then? Test could fail even when the actual error is what we want it to be?

Reply:
its by design, since if something doesnt exist, the function should return `os.ErrNotExist` (or `fmt.Errorf("...: %w", os.ErrNotExist)`), or whatever the agreed upon error is.
otherwise, we would have:
```
var ErrFileNotFound = errors.New("not found")
var ErrHandleNotFound = errors.New("not found")

errors.Is(ErrFileNotFound, ErrHandleNotFound)
```

also, (most of) hcsshim uses `errors.Is` instead of string matching (since the latter is bad practice and linters will complain), so this way tests match prod code